Add io.github.benjamimgois.goverlay

benjamimgois opened the pull request:
<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [X] Please describe the application briefly. < Goverlay is an easy graphical interface to configure MangoHud, vkBasalt, and OptiScaler >
- [X] Please attach a video showcasing the application on Linux using the Flatpak. < https://github.com/user-attachments/assets/22220858-afb8-4e58-8c5c-74bb4f858a99 >
- [X] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [X] I am an author of the project.
     
<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

benjamimgois commented:
thanks for the feedback @hfiguiere ! it's my first flathub submission so i still have things to learn. I'll try to work on it when i got home

bbhtt commented on io.github.benjamimgois.goverlay.yml:
```diff
@@ -0,0 +1,283 @@
+        export PATH=/app/bin:$PATH
+        export Qt6Pas_LibDir=/app/lib
+        # Detect architecture for FPC units path
+        FPC_ARCH=$(uname -m)
```
use $FLATPAK_ARCH no need to uname

hfiguiere replied:
lazarus is also in the freepascal SDK. There is no need to build either.

bbhtt commented on io.github.benjamimgois.goverlay.yml:
```diff
@@ -0,0 +1,283 @@
+    cleanup:
+      - /include
+      - /lib/pkgconfig
+      - '*.a'
```
these should be in global cleanup

benjamimgois commented:
Hi @hfiguiere and @bbhtt,

I've addressed all the feedback from the review:

-  Using org.freedesktop.Sdk.Extension.freepascal as requested
-  Changed source type from 'dir' to 'git' with proper tag and commit
-  Removed add-extensions for MangoHud and vkBasalt (now bundled)
-  Using portal notification (org.freedesktop.portal.Notification)
-  Removed /sys filesystem permission
-  Moved cleanup section above modules
-  Added Qt6 Wayland for complete Wayland support
-  Included Breeze Icons for better visual consistency

The manifest has been tested locally and builds successfully.

hfiguiere commented on io.github.benjamimgois.goverlay.yml:
```diff
@@ -0,0 +1,230 @@
+app-id: io.github.benjamimgois.goverlay
+runtime: org.freedesktop.Platform
```
and where will it find Qt now?

hfiguiere commented on io.github.benjamimgois.goverlay.yml:
```diff
@@ -0,0 +1,191 @@
+  - --socket=fallback-x11
+
+  # Desktop notifications
+  - --talk-name=org.freedesktop.Notifications
```
this should not be needed.

Twig6943 replied:
Removed notification perms

hfiguiere commented on io.github.benjamimgois.goverlay.yml:
```diff
@@ -0,0 +1,191 @@
+    buildsystem: simple
+    build-commands:
+      - |
+        # Extract .deb package properly
```
why isn't this built from source?

Twig6943 replied:
Its a leftover from benja's version of the manifest. I thought it'd be ok to extract from .deb, I'll get it building from src tomorrow. (its 2 am here rn lol)

Twig6943 commented on io.github.benjamimgois.goverlay.yml:
```diff
@@ -0,0 +1,191 @@
+  - --filesystem=xdg-config/vkBasalt:rw
+  - --filesystem=xdg-config/goverlay:rw
+  - --filesystem=xdg-data/goverlay:rw
+  - --filesystem=~/fgmod:rw
```
I don't get what you mean (Sorry)

hfiguiere replied:
use `--persist` for this one.

bbhtt commented:
bot, build

Twig6943 commented:
> the name in metainfo is simply "App" instead of Goverlay. Also add some more description, features etc.

On it

Twig6943 commented:
Updated; 
[to make it use the latest release + added steam permission for flatpak steam](https://github.com/benjamimgois/flathub/commit/6a3d01125604ff50456b5186a43c1ea5c5951c86)

[Updated metainfo](https://github.com/benjamimgois/goverlay/commit/097b2082164e9f78f6e443d0e6497e889b945f1f)

The steam permission requires another exception.json PR so [here's that](https://github.com/flathub-infra/flatpak-builder-lint/pull/815)

I'm going to check if everything went smoothly with bot build

Twig6943 commented:
bot, build

Twig6943 commented:
Oh yea I forgot it failed because of the pr being unmerged my bad;

<img width="871" height="121" alt="image" src="https://github.com/user-attachments/assets/96d56e77-51c4-49cd-84e6-493f8443dd99" />
